src/main/java/org/sonarqube/gradle/SourceCollector.java

```diff
@@ -92,7 +98,12 @@ public Set<Path> getCollectedSources() {
 
   private final Set<Path> collectedSources = new HashSet<>();
 
-  public SourceCollector(Set<Path> existingSources, Set<Path> directoriesToIgnore, Set<Path> excludedFiles, boolean shouldCollectJavaAndKotlinSources) {
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  public SourceCollector(Path root, Set<Path> existingSources, Set<Path> directoriesToIgnore, Set<Path> excludedFiles, boolean shouldCollectJavaAndKotlinSources) {
```

Comment:
If we create an object using the builder pattern, is there a reason to keep the constructor public?